### ISSUE TITLE
UnboundLocalError: local variable 'detection' referenced before assignment

### ISSUE THREAD

**Issue by kgs** [gaubert/gmvault#199]
I am using gmvault 1.9 while doing sync, from time to time I get following error:

```
Process email num 5979 (imap_id:26570) from 2015-02.

Error: local variable 'detection' referenced before assignment.

=== Exception traceback ===
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/gmv/gmv_cmd.py", line 741, in run
    self._sync(args, credential)
  File "/usr/local/lib/python2.7/dist-packages/gmv/gmv_cmd.py", line 661, in _sync
    restart = args['restart'], emails_only = args['emails_only'], chats_only = args['chats_only'])
  File "/usr/local/lib/python2.7/dist-packages/gmv/gmvault.py", line 607, in sync
    self._sync_emails(imap_req, compress = compress_on_disk, restart = restart)
  File "/usr/local/lib/python2.7/dist-packages/gmv/gmvault.py", line 570, in _sync_emails
    imap_ids = self._common_sync(timer, "email", imap_req, compress, restart)
  File "/usr/local/lib/python2.7/dist-packages/gmv/gmvault.py", line 528, in _common_sync
    handle_sync_imap_error(error, the_id, self.error_report, self.src) #do everything in this handler
  File "/usr/local/lib/python2.7/dist-packages/gmv/gmvault.py", line 154, in handle_sync_imap_error
    raise the_exception
UnboundLocalError: local variable 'detection' referenced before assignment
````

**Comment by gaubert** [comment]
@kgs chardet is used to find the encoding and it can return None (even if this is not indicated in the API). I have fixed that issue in v1.9.1.

**Comment by kgs** [comment]
@gaubert yes, I can test it, but I am not sure if I can trigger this behaviour, although i can try :)

**Comment by gaubert** [comment]
@kgs Ok great I will let you know in few days when it is ready. I want to modify a bit how I use chardet because it isn't really reliable.

**Comment by GeoffMaciolek** [comment]
I'm experiencing similar behavior (1.9), except it looks like it's a bit more explicit on mine?

<pre>Process email num 59156 (imap_id:186536) from 2013-11.
Expected a bytes object, not a unicode object
Error: local variable 'detection' referenced before assignment.

=== Exception traceback ===
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/gmv/gmv_cmd.py", line 741, in run
    self._sync(args, credential)
  File "/usr/lib/python2.7/site-packages/gmv/gmv_cmd.py", line 661, in _sync
    restart = args['restart'], emails_only = args['emails_only'], chats_only = args['chats_only'])
  File "/usr/lib/python2.7/site-packages/gmv/gmvault.py", line 607, in sync
    self._sync_emails(imap_req, compress = compress_on_disk, restart = restart)
  File "/usr/lib/python2.7/site-packages/gmv/gmvault.py", line 570, in _sync_emails
    imap_ids = self._common_sync(timer, "email", imap_req, compress, restart)
  File "/usr/lib/python2.7/site-packages/gmv/gmvault.py", line 528, in _common_sync
    handle_sync_imap_error(error, the_id, self.error_report, self.src) #do everything in this handler
  File "/usr/lib/python2.7/site-packages/gmv/gmvault.py", line 154, in handle_sync_imap_error
    raise the_exception
UnboundLocalError: local variable 'detection' referenced before assignment

=== End of Exception traceback ===</pre>

What's interesting is that I've used gmvault in the past on that same mail content - though I *think* I was running 1.8.1 with the patch from #147  - related?

**Comment by gaubert** [comment]
@GeoffMaciolek fixed in 1.9.1

**Comment by johndangerstorey** [comment]
I just downloaded and implemented version 1.9.1 and it's giving me a very similar error of
`UnboundLocalError: local variable 'res' referenced before assignment`

Here is the output

```
ushing email body with id 1414150177260942940.
Error: local variable 'res' referenced before assignment.

=== Exception traceback ===
Traceback (most recent call last):
  File "/Users/gaubert/Dev/projects/gmvault-1.8.2/build/gmvault/out00-PYZ.pyz/gmv.gmv_cmd", line 750, in run
  File "/Users/gaubert/Dev/projects/gmvault-1.8.2/build/gmvault/out00-PYZ.pyz/gmv.gmv_cmd", line 625, in _restore
  File "/Users/gaubert/Dev/projects/gmvault-1.8.2/build/gmvault/out00-PYZ.pyz/gmv.gmvault", line 901, in restore
  File "/Users/gaubert/Dev/projects/gmvault-1.8.2/build/gmvault/out00-PYZ.pyz/gmv.gmvault", line 1142, in restore_emails
  File "/Users/gaubert/Dev/projects/gmvault-1.8.2/build/gmvault/out00-PYZ.pyz/gmv.gmvault", line 84, in handle_restore_imap_error
UnboundLocalError: local variable 'res' referenced before assignment

=== End of Exception traceback ===
```

Any ideas?

**Comment by gaubert** [comment]
@johndangerstorey ok tried to fix it in v1.9.2
